replica/table.cc

```diff
@@ -2714,22 +2709,22 @@ table::stream_view_replica_updates(shared_ptr<db::view::view_update_generator> g
             s,
             std::move(m),
             timeout,
-            as_mutation_source_excluding(excluded_sstables),
+            as_mutation_source_excluding_staging(),
             tracing::trace_state_ptr(),
             *_config.streaming_read_concurrency_semaphore,
             query::partition_slice::option_set::of<query::partition_slice::option::bypass_cache>());
 }
 
 mutation_source
-table::as_mutation_source_excluding(std::vector<sstables::shared_sstable>& ssts) const {
-    return mutation_source([this, &ssts] (schema_ptr s,
+table::as_mutation_source_excluding_staging() const {
+    return mutation_source([this] (schema_ptr s,
                                    reader_permit permit,
                                    const dht::partition_range& range,
                                    const query::partition_slice& slice,
                                    tracing::trace_state_ptr trace_state,
                                    streamed_mutation::forwarding fwd,
                                    mutation_reader::forwarding fwd_mr) {
-        return this->make_reader_v2_excluding_sstables(std::move(s), std::move(permit), ssts, range, slice, std::move(trace_state), fwd, fwd_mr);
+        return this->make_reader_v2_excluding_staging(std::move(s), std::move(permit), range, slice, std::move(trace_state), fwd, fwd_mr);
```

Comment:
Q: isn't  sstable_set = staging + main? Can't we read just from main_sstable_set?

Reply:
I considered it and I gave it lots of thought, but I stepped away from it because staging going to maintenance set is not an invariant. Some streaming operations may bypass off-strategy, therefore staging won't go into maintenance set. Additionally, there might be some sstables in maintenance set which was moved away from staging and not reshaped yet, and should be considered for view building.

Reply:
No, it's main + maintenance .
@raphaelsc and I discussed adding a third subset which is staging but it may make other things more complicated.

Reply:
A small correction: I think now all streaming operations use off-strategy. We could make staging sstables belonging only to maintenance set an invariant, refusing to load staging sstables into main. But we could still suffer from the fact that some sstables in maintenance are no longer in staging condition and should be considered by view building. So the predicate applied against compound set is the safer solution we have, as it considers all sets and only include sstables that satisfy a criteria.